### PR TITLE
proxy: Enable stricter/pedantic clippy checks

### PR DESCRIPTION
**Pull request by cloneable**
Create a list of currently allowed exceptions that should be reduced over time.

### CODE REVIEW

File: proxy/src/lib.rs [hunk 1]
```diff
@@ -1,4 +1,109 @@
...
+// rustc lints/lint groups
+// https://doc.rust-lang.org/rustc/lints/groups.html
+#![deny(
+    deprecated,
+    future_incompatible,
+    // TODO: consider let_underscore
+    nonstandard_style,
+    rust_2018_idioms,
+    rust_2021_compatibility,
+    rust_2024_compatibility
```
Comment (cloneable) [inline proxy/src/lib.rs:10]: Are these three in any way mutually exclusive? They don't seem to overlap in what they enable nor do any of their lints conflict.

File: proxy/src/lib.rs [hunk 2]
```diff
@@ -1,4 +1,109 @@
...
+)]
+#![warn(clippy::all, clippy::pedantic, clippy::nursery, clippy::cargo)]
```
Comment (conradludgate) [inline proxy/src/lib.rs:12]:
```suggestion
#![warn(clippy::all, clippy::pedantic, clippy::cargo)]
```

I don't mind pedantic too much, but nursery is probably too inconsistent.

File: proxy/src/lib.rs (hunk 1 above)
Comment (conradludgate) [reply proxy/src/lib.rs:10]: Yeah, let's remove the first two.

Comment (cloneable) [reply proxy/src/lib.rs:8]: Done.

File: proxy/src/lib.rs (hunk 2 above)
Comment (cloneable) [reply proxy/src/lib.rs]: Done.

I also removed the exceptions.

**Review by conradludgate — APPROVED** [review]

**Pull request merged**